examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -147,21 +149,60 @@ UserConsentSubject OTAProviderExample::GetUserConsentSubject(const app::CommandH
     return subject;
 }
 
+bool OTAProviderExample::ParseOTAHeader(const char * otaFilePath, OTAImageHeader & header)
+{
+    OTAImageHeaderParser parser;
+    uint8_t otaFileContent[kOtaHeaderMaxSize];
+    ByteSpan buffer(otaFileContent);
+
+    std::ifstream otaFile(otaFilePath, std::ifstream::in);
+    if (!otaFile.is_open() || !otaFile.good())
+    {
+        ChipLogError(SoftwareUpdate, "Error opening OTA image file: %s", otaFilePath);
+        return false;
+    }
+
+    otaFile.read(reinterpret_cast<char *>(otaFileContent), kOtaHeaderMaxSize);
+    if (!otaFile.good() || otaFile.eof())
```

Comment:
`eof()` doesn't mean that the read failed. It can happen for some "theoretic", very small images.

Reply:
So this was based on the header size of 128 so I figured any application must be bigger than that :) With your comment above, I'll remove the `eof` check here.